trpcultivate_germplasm/tests/src/Kernel/TripalImporter/GermplasmAccessionImporter/GermplasmAccessionImporterRunTest.php

```diff
@@ -0,0 +1,375 @@
+  }
+
+  /**
+   * Tests focusing on the Germplasm Accession Importer run() function
+   * using an example file that should specifically cause an exception
+   * to occur due to the following cases:
+   *   1. A non-existant organism in the database
+   *   2. Attempt to insert a duplicate stock accession
+   *
+   * @group germ_accession_importer
+   */
+  public function testGermplasmAccessionImporterRunIncomplete() {
+
+    // Test for a non-existant file
+    $non_existant_file = __DIR__ . '/does_not_exist.txt';
+
+    $genus = 'Sally';
+    $run_args = ['genus_name' => $genus];
+    $file_details = ['file_local' => $non_existant_file];
+
+    $this->importer->createImportJob($run_args, $file_details);
+    $this->importer->prepareFiles();
+
+    $exception_caught = FALSE;
+    try {
+      $this->importer->run();
+    } catch ( \Exception $e ) {
+      $exception_caught = TRUE;
+    }
+    $this->assertStringContainsString("File does not exist:", $e->getMessage(), "Expected an exception message that file \"does_not_exist.txt\", does not, in fact, exist.");
+    $this->assertTrue($exception_caught, "Did not catch exception for a non-existant file as input.");
+
+    $incomplete_example_file = __DIR__ . '/../../../Fixtures/incomplete_example.txt';
+
+    // Test for a non-existant organism
+    $genus = 'Sally';
+    $run_args = ['genus_name' => $genus];
+    $file_details = ['file_local' => $incomplete_example_file];
+
+    $this->importer->createImportJob($run_args, $file_details);
+    $this->importer->prepareFiles();
+
+    $exception_caught = FALSE;
+    ob_start();
+    try {
+      $this->importer->run();
+    } catch ( \Exception $e ) {
+      $exception_caught = TRUE;
+    }
+    $printed_output = ob_get_clean();
+    $this->assertStringContainsString("Could not find an organism", $printed_output, "Expected an error that an organism in the file 'incomplete_example.txt' could not be found in the database.");
+    $this->assertTrue($exception_caught, "Did not catch exception for trying to insert germplasm with a non-existant organism.");
+
+    // Test for existing organism, but non-existing stockID
+    $genus = 'Tripalus';
+    $run_args = ['genus_name' => $genus];
+
+    $this->importer->createImportJob($run_args, $file_details);
+    $this->importer->prepareFiles();
+
+    $exception_caught = FALSE;
+    ob_start();
+    try {
+      $this->importer->run();
+    } catch ( \Exception $e ) {
+      $exception_caught = TRUE;
+    }
+    $printed_output = ob_get_clean();
+    $this->assertStringContainsString('A stock already exists for accession "T1" but with a germplasm name of "Test1" which does not match the input file.', $printed_output, "Expected an error that a stock accession 'T1' already exists in the file 'incomplete_example.txt'");
+    $this->assertTrue($exception_caught, "Did not catch exception for trying to insert duplicate stock accession numbers.");
```

Comment:
To catch what I'm seeing in my recent review, you want to add a query + assert here that checks that no stock records have been inserted into the chado database.
This test passes because I do see the warning :-) but it should fail because the database did not get rolled back.

PS. My terminal output:
```
docker exec -it germG2.2  drush trp-run-jobs --job_id=1 --username=drupaladmin --root=/var/www/drupal/web

2024-01-19 18:57:39
Tripal Job Launcher
Running as user 'drupaladmin'
-------------------
2024-01-19 18:57:39: Job ID 1.
2024-01-19 18:57:39: Calling: tripal_run_importer(34)
Running 'Tripal Cultivate: Germplasm Accessions' importer
NOTE: Loading of this file is performed using a database transaction. If it fails or is terminated prematurely then all insertions and updates are rolled back and will not be found in the database
Percent complete: 0%. Memory: 34,018,024 bytes.
Percent complete: 1.12 %. Memory: 34,026,848 bytes.
Percent complete: 3.21 %. Memory: 34,026,920 bytes.
Percent complete: 4.33 %. Memory: 34,026,984 bytes.
Percent complete: 32.96 %. Memory: 34,027,264 bytes.
Percent complete: 56.56 %. Memory: 34,027,392 bytes.
Inserting "CDC Ch@d".
Percent complete: 74.58 %. Memory: 34,087,336 bytes.
Inserting "USASK Ken".
Percent complete: 100.00 %. Memory: 34,087,456 bytes.
Percent complete: 100.00 %. Memory: 34,087,816 bytes.
ERROR: A stock already exists for accession "GERM:00001" but with a germplasm name of "CDC Ch@d" which does not match the input file.
[site http://default] [TRIPAL] ERROR: A stock already exists for accession "GERM:00001" but with a germplasm name of "CDC Ch@d" which does not match the input file.
Percent complete: 100.00 %. Memory: 34,087,592 bytes.
ERROR: The database transaction was not commited due to the presence of one or more errors. Please fix all errors and try the import again.
[site http://default] [TRIPAL] ERROR: The database transaction was not commited due to the presence of one or more errors. Please fix all errors and try the import again.
```

Reply:
Thanks Lacey! Query & assert was added in commit 03b4faef250f4d01dc053e0495b71dab1e74a8a7